pkgs/native_toolchain_c/lib/src/native_toolchain/clang.dart

```diff
@@ -68,6 +73,11 @@ final Tool lld = Tool(
         wrappedResolver: clang.defaultResolver!,
         relativePath: Uri.file(OS.current.executableFileName('ld.lld')),
       ),
+      RelativeToolResolver(
```

Comment:
Can we add a `keepIf` to avoid categorizing a non `ld.lld` (the one from GCC or apple clang) to be be found here.